File: lista ads/q10.py
a = [10, 5, -17, 20, 50, -1, 3, -30, 10]
maior_soma = a[0] 

for x in range(len(a)):
    soma_atual = 0
    for y in range(x, len(a)):
        soma_atual += a[y] 
        if soma_atual > maior_soma:
            maior_soma = soma_atual

print(maior_soma)  

File: lista ads/java/Q10.java
public class Q10 {
    public static void main(String[] args) {
        
        int array_A[] = {10, 5, -17, 20, 50, -1, 3, -30, 10};
        int somaMax = 0;

        for(int i = 0; i < array_A.length; i++) {
            int somaAtual = 0;
            for(int j = i; j < array_A.length; j++) {
                somaAtual += array_A[j];

                if(somaAtual > somaMax) {
                    somaMax = somaAtual;
                }
            }
        }

        System.out.println(somaMax);

    }
}
